src/spend.rs

```diff
@@ -374,14 +390,27 @@ fn select_coins_for_spend(
     // By now, selection is complete and we can check how much change to give according to our policy.
     let drain = change_policy(&selector, target);
     let change_amount = bitcoin::Amount::from_sat(drain.value);
-    Ok((
+    // Max available change is given by the excess when adding a change output with zero value.
+    let drain_novalue = bdk_coin_select::Drain {
+        weights: drain_weights,
+        value: 0,
+    };
+    let max_change_amount = bitcoin::Amount::from_sat(
         selector
+            .excess(target, drain_novalue)
+            .max(0) // negative excess would mean insufficient funds to pay for change output
```

Comment:
So this can only happen if we've got a selection without change such as adding a change output would decrease the feerate below the target? Any other error here would be a logic error because at this point we do indeed have enough funds?

Reply:
That's right, the selected coins here have enough value without a change output and so the negative excess would mean that the selected coins don't have enough value to pay the extra weight for the change output at the target feerate.